lib/perl/Genome/Model/ReferenceAlignment/Command/LibraryDuplicationRates.pm

```diff
@@ -0,0 +1,63 @@
+package Genome::Model::ReferenceAlignment::Command::LibraryDuplicationRates;
+
+use strict;
+use warnings;
+
+class Genome::Model::ReferenceAlignment::Command::LibraryDuplicationRates {
+    is => 'Command::V2',
+    has => [
+        builds => {
+            is_many => 1,
+            is => 'Genome::Model::Build::ReferenceAlignment',
+            shell_args_position => 1,
+            doc => 'List of builds to report on.',
+        },
+    ],
+};
+
+sub help_brief {
+    'A command to print Picard MarkDuplicates duplication metrics for a reference alignment build.'
+}
+
+sub help_synopsis {
+    'Dump reference alignment build duplication metrics to standard out in tab-delimited format.'
+}
+
+sub help_detail{
+    return <<"EOS"
+All reference alignment builds currently use Picard MarkDuplicates to flag PCR duplicates after merging the individual instrument data alignments.
+One output from Picard MarkDuplicates is a metrics file that provides a per library breakdown of PCR and optical duplicate reads.  This tool simply parses the metrics file and dumps those metrics along with basic build/model information in tab-delimited form.
+EOS
+}
+
+sub execute {
+    my $self = shift;
+
+    my $writer;
```

Comment:
Would it be better to store a hash for each build, the at the end, uniq the keys of the data for the headers, open the writer, and write the entries? This eliminates the need to check if the writer exists, and would prevent the output method from possibly being called on the undefined writer below.

```perl
# @data = { BUILD_ID => 1, ... } ...
if ( not @data ) { # necessary?
    $self->warning_message('No mark dup found...');
    return 1; # command ran OK
}
my @headers = List::MoreUtils::uniq(sort map { keys %$_ } @data);
my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
    separator => "\t",
    headers => \@headers,
);
for ( @data ) { $writer->write_one($_); }
```